onnxruntime/core/framework/kernel_registry_manager.h

```diff
@@ -17,46 +18,56 @@ class KernelRegistry;
 class OpKernel;
 class SessionState;
 
-enum class KernelRegistryPriority {
-  HighPriority,
-  LowPriority
-};
-
 // Kernel registries' manager.
 // There're 2 kinds of kernel registries with priority from high to low as below,
 // 1. Custom execution provider type specific kernel registries.
 // 2. common execution provider type specific kernel registries.
 // The 1st and 2nd ones are shared across sessions.
+
+//This class is thread safe
 class KernelRegistryManager {
  public:
   KernelRegistryManager() = default;
 
-  void RegisterKernels(const ExecutionProviders& execution_providers,
-                       KernelRegistryPriority priority = KernelRegistryPriority::LowPriority);
+  //Register kernels from providers
+  void RegisterKernels(const ExecutionProviders& execution_providers);
 
-  void RegisterKernelRegistry(std::shared_ptr<KernelRegistry> kernel_registry, KernelRegistryPriority priority);
+  // The registry passed in this function has highest priority than anything already in this KernelRegistryManager,
+  // and anything registered from RegisterKernels
+  // For example, if you do:
+  // RegisterKernels(providers)
+  // RegisterKernelRegistry(A);
+  // RegisterKernelRegistry(B);
+  // Then B > A > providers
+  void RegisterKernelRegistry(std::shared_ptr<KernelRegistry> kernel_registry);
 
   Status CreateKernel(const onnxruntime::Node& node,
                       const IExecutionProvider& execution_provider,
                       const SessionState& session_state,
                       /*out*/ std::unique_ptr<OpKernel>& op_kernel) const;
 
+  //This function assumes the node is already assigned to an execution provider
+  //Don't call this function before graph partition is done
   Status SearchKernelRegistry(const onnxruntime::Node& node,
                               /*out*/ const KernelCreateInfo** kernel_create_info) const;
 
+  bool HasImplementationOf(const Node& node, const std::string& provider_type) const;
+
   // Get all kernel registries. There are no nullptr entries.
```

Comment:
>Get all kernel registries [](start = 5, length = 25)

This comment doesn't really describe the code does it? Seems like this returns all custom registries, and the registry for the given provider type.